src/pymmcore_widgets/_mda/_grid_widget.py

```diff
@@ -1,228 +1,607 @@
+    mmcore : CMMCorePlus | None
+        Optional [`pymmcore_plus.CMMCorePlus`][] micromanager core.
+        By default, None. If not specified, the widget will use the active
+        (or create a new)
+        [`CMMCorePlus.instance`][pymmcore_plus.core._mmcore_plus.CMMCorePlus.instance].
+    """
+
+    valueChanged = Signal(object)
+
+    def __init__(
+        self,
+        parent: QWidget | None = None,
+        *,
+        current_stage_pos: tuple[float | None, float | None] = (None, None),
+        mmcore: CMMCorePlus | None = None,
+    ) -> None:
+        super().__init__(parent=parent)
 
-        self._create_gui()
+        self._mmc = mmcore or CMMCorePlus.instance()
 
-        self._update_info_label()
+        self._current_stage_pos = current_stage_pos
 
-    def _create_gui(self) -> None:
         layout = QVBoxLayout()
-        layout.setSpacing(5)
+        layout.setSpacing(10)
         layout.setContentsMargins(10, 10, 10, 10)
         self.setLayout(layout)
 
-        self.setMaximumHeight(200)
+        tab = self._create_tab()
+        layout.addWidget(tab)
+
+        overlap_and_size = self._create_overlap_and_ordermode()
+        layout.addWidget(overlap_and_size)
 
-        grid_settings = self._create_row_cols_overlap_group()
-        layout.addWidget(grid_settings)
+        move_to = self._create_move_to_widget()
+        layout.addWidget(move_to)
 
-        button = self._create_generate_list_button()
+        label_info = self._create_label_info()
+        layout.addWidget(label_info)
+
+        button = self._create_add_button()
         layout.addWidget(button)
 
-    def _create_row_cols_overlap_group(self) -> QGroupBox:
-        group = QGroupBox(title="Grid Parameters")
-        group.setSizePolicy(
-            QSizePolicy(QSizePolicy.Policy.Minimum, QSizePolicy.Policy.Fixed)
-        )
+        self.setFixedHeight(self.sizeHint().height())
+
+        self._update_info()
+
+        self._mmc.events.systemConfigurationLoaded.connect(self._update_info)
+        self._mmc.events.pixelSizeChanged.connect(self._update_info)
+
+        self.destroyed.connect(self._disconnect)
+
+    def _create_tab(self) -> QWidget:
+        wdg = QWidget()
+        layout = QVBoxLayout()
+        layout.setSpacing(0)
+        layout.setContentsMargins(0, 0, 0, 0)
+        wdg.setLayout(layout)
+        self.tab = QTabWidget()
+        self.tab.setSizePolicy(QSizePolicy.Policy.Expanding, QSizePolicy.Policy.Fixed)
+
+        rowcol = self._create_row_cols_wdg()
+        self.tab.addTab(rowcol, "Rows x Columns")
+
+        edges = self._create_edges_grid_wdg()
+        self.tab.addTab(edges, "Grid from Edges")
+
+        corners = self._create_corners_grid_wdg()
+        self.tab.addTab(corners, "Grid from Corners")
+
+        layout.addWidget(self.tab)
+
+        self.tab.currentChanged.connect(self._update_info)
+        return wdg
+
+    def _create_row_cols_wdg(self) -> QWidget:
```

Comment:
i really don't like these methods that exist just to create an return specialized widget... instead, make a new class, and instantiate it:
```python
class _RowsColsWdg(QWidget):
    def __init__(self):
        self.setLayout(QGridLayout())
        ...  # etc
```